src/Microsoft.Net.Http.Headers/ContentDispositionHeaderValue.cs

```diff
@@ -69,13 +69,14 @@ public IList<NameValueHeaderValue> Parameters
 
         public StringSegment Name
         {
-            get { return GetName(NameString); }
+            get { return HeaderUtilities.RemoveQuotes(GetName(NameString)); }
```

Comment:
The actual data is stored in the Parameters collection and is directly modifiable. This is just a convenience accessor. Any attempt at caching would have to check that the underlying version hadn't changed.